psutil/_psutil_aix.c

```diff
@@ -134,45 +135,81 @@ psutil_proc_basic_info(PyObject *self, PyObject *args) {
 
 
 /*
- * Return process name and args as a Python tuple.
+ * Return process name as a Python string.
  */
 static PyObject *
-psutil_proc_name_and_args(PyObject *self, PyObject *args) {
+psutil_proc_name(PyObject *self, PyObject *args) {
     int pid;
     char path[100];
     psinfo_t info;
     const char *procfs_path;
-    PyObject *py_name = NULL;
-    PyObject *py_args = NULL;
-    PyObject *py_retlist = NULL;
 
     if (! PyArg_ParseTuple(args, "is", &pid, &procfs_path))
         return NULL;
     sprintf(path, "%s/%i/psinfo", procfs_path, pid);
     if (! psutil_file_to_struct(path, (void *)&info, sizeof(info)))
         return NULL;
 
-    py_name = PyUnicode_DecodeFSDefault(info.pr_fname);
-    if (!py_name)
+    return PyUnicode_DecodeFSDefault(info.pr_fname);
+}
+
+
+/*
+ * Return process command line arguments as a Python list
+ */
+static PyObject *
+psutil_proc_args(PyObject *self, PyObject *args) {
+    int pid;
+    PyObject *py_retlist = PyList_New(0);
+    PyObject *py_arg = NULL;
+    struct procsinfo procbuf;
+    long arg_max;
+    char *argbuf = NULL;
+    char *curarg = NULL;
+    int ret;
+
+    if (py_retlist == NULL)
+        return NULL;
+    if (!PyArg_ParseTuple(args, "i", &pid))
         goto error;
-    py_args = PyUnicode_DecodeFSDefault(info.pr_psargs);
-    if (!py_args)
+    arg_max = sysconf(_SC_ARG_MAX);
+    argbuf = malloc(arg_max);
+    if (!argbuf)
```

Comment:
set Py_ErrorNoMemory before “goto error” (sorry I am on my phone and i don’t remember the exact name of the cPython api to set the memory error exception)